lib/Dialect/TritonGPU/Transforms/AccelerateMatmul.cpp

```diff
@@ -334,6 +334,52 @@ class BlockedToMMA : public mlir::RewritePattern {
 };
 } // namespace
 
+static Value promoteOperand(OpBuilder &builder, Location loc, Value operand,
+                            Type promotedType) {
+  Type tensorPromotedType =
+      operand.getType().cast<RankedTensorType>().cloneWith(std::nullopt,
+                                                           promotedType);
+  return builder.create<tt::FpToFpOp>(loc, tensorPromotedType, operand);
```

Comment:
The type of input operands might *not* be a floating point type. If that happens, creating a `tt::FpToFpOp` would be incorrect. I believe this function should handle operands that have an integral type because `tt.dot` allows that kind of operands as exemplified in https://github.com/openai/triton/blob/ded624282e67e5f58db332380e6ff088f276d534/python/test/unit/language/test_line_info.py#L70